coderd/database/pubsub/watchdog_test.go

```diff
@@ -4,36 +4,48 @@ import (
 	"testing"
 	"time"
 
-	"github.com/benbjohnson/clock"
 	"github.com/stretchr/testify/require"
 
 	"cdr.dev/slog"
 	"cdr.dev/slog/sloggers/slogtest"
+	"github.com/coder/coder/v2/clock"
 	"github.com/coder/coder/v2/coderd/database/pubsub"
 	"github.com/coder/coder/v2/testutil"
 )
 
 func TestWatchdog_NoTimeout(t *testing.T) {
 	t.Parallel()
-	ctx := testutil.Context(t, time.Hour)
+	ctx := testutil.Context(t, testutil.WaitShort)
 	mClock := clock.NewMock()
-	start := time.Date(2024, 2, 5, 8, 7, 6, 5, time.UTC)
-	mClock.Set(start)
 	logger := slogtest.Make(t, &slogtest.Options{IgnoreErrors: true}).Leveled(slog.LevelDebug)
 	fPS := newFakePubsub()
+
+	// trap the ticker and timer.Stop() calls
+	pTrap := mClock.Trap().NewContextTicker("publish")
+	defer pTrap.Close()
+	sTrap := mClock.Trap().TimerStop("subscribe")
+	defer sTrap.Close()
+
 	uut := pubsub.NewWatchdogWithClock(ctx, logger, fPS, mClock)
 
+	// wait for the ticker to be created so that we know it starts from the
+	// right baseline time.
+	pc, err := pTrap.Wait(ctx)
+	require.NoError(t, err)
+	require.Equal(t, 15*time.Second, pc.Duration)
+	pc.Release()
```

Comment:
The require(s) above may result in release never being called. Just pointing this out in case it matters (e.g. will we leave goroutines behind, block in some other defer, etc?).